key-manager/src/main/scala/org/bitcoins/keymanager/WalletStorage.scala

```diff
@@ -316,40 +300,59 @@ object WalletStorage extends KeyManagerLogger {
       passphraseOpt: Option[AesPassword]): Either[
     ReadMnemonicError,
     DecryptedSeedState] = {
-    val decryptedEither: Either[ReadMnemonicError, DecryptedSeedState] =
-      passphraseOpt match {
-        case Some(passphrase) =>
-          val encryptedEither = readEncryptedMnemonicFromDisk(seedPath)
-
-          encryptedEither.flatMap { encrypted =>
-            encrypted.toMnemonic(passphrase) match {
-              case Failure(_) =>
-                encrypted.toExtPrivKey(passphrase) match {
-                  case Failure(exc) =>
-                    logger.error(s"Error when decrypting $encrypted: $exc")
-                    Left(ReadMnemonicError.DecryptionError)
-                  case Success(xprv) =>
+    import MnemonicJsonKeys._
```

Comment:
Is there any way we can refactor this to be more readable? This is really starting to grow into something that is difficult to analyze if it's correct

Reply:
Moved into a separate function. Should be easier to read now